compiler/src/dotty/tools/dotc/Run.scala

```diff
@@ -41,7 +42,23 @@ class Run(comp: Compiler, ictx: Context) extends ImplicitRunInfo with Constraint
    */
   protected[this] def rootContext(implicit ctx: Context): Context = {
     ctx.initialize()(ctx)
-    ctx.base.setPhasePlan(comp.phases)
+
+    val actualPhases = if (ctx.settings.scalajs.value) {
+      // Remove phases that Scala.js does not want
+      comp.phases.mapConserve(_.filter {
+        case _: transform.FunctionalInterfaces => false
+        case _ => true
+      }).filter(_.nonEmpty)
+    } else {
+      // Remove Scala.js-related phases
+      comp.phases.mapConserve(_.filter {
+        case _: sjs.GenSJSIR => false
+        case _ => true
+      }).filter(_.nonEmpty)
+    }
+
+    ctx.base.setPhasePlan(actualPhases)
```

Comment:
It would be better if `sjs` defines a subclass of `Compiler` where it does this filtering and adds the  [new sjs.GenSJSIR](https://github.com/lampepfl/dotty/pull/5143/files#diff-baf9e45272584379584b8c1e5467ddf5R117) phase.

Reply:
But then if we want to compile to SJSIR from TASTY, I would need a subclass of both the `SJSIRCompiler` and the `TASTYCompiler`, and then things start to get messy.

Reply:
Moved the tests to the phases themselves, as discussed during the meeting.